tests/test_synced_collection.py

```diff
@@ -0,0 +1,587 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import pytest
+import uuid
+import os
+import json
+from tempfile import TemporaryDirectory
+from collections.abc import MutableMapping
+from collections.abc import MutableSequence
+
+from signac.core.synced_list import SyncedCollection
+from signac.core.jsoncollection import JSONDict
+from signac.core.jsoncollection import JSONList
+from signac.errors import InvalidKeyError
+from signac.errors import KeyTypeError
+
+try:
+    import numpy
+    NUMPY = True
+except ImportError:
+    NUMPY = False
+
+FN_JSON = 'test.json'
+
+
+@pytest.fixture
+def testdata():
+    return str(uuid.uuid4())
+
+
+class TestSyncedCollectionBase():
+
+    @pytest.fixture(autouse=True)
+    def synced_collection(self):
+        self._tmp_dir = TemporaryDirectory(prefix='jsondict_')
+        self._fn_ = os.path.join(self._tmp_dir.name, FN_JSON)
+        yield
+        self._tmp_dir.cleanup()
+
+    def test_from_base(self):
+        print(JSONDict.backend)
+        sd = SyncedCollection.from_base(filename=self._fn_,
+                                        data={'a': 0}, backend='signac.core.jsoncollection')
+        assert isinstance(sd, JSONDict)
+        assert 'a' in sd
+        assert sd['a'] == 0
+
+        # invalid input
+        with pytest.raises(ValueError):
+            SyncedCollection.from_base(data={'a': 0}, filename=self._fn_)
+
+
+class TestJSONDict():
+
+    _write_concern = False
+
+    @pytest.fixture(autouse=True)
+    def synced_dict(self):
+        self._tmp_dir = TemporaryDirectory(prefix='jsondict_')
+        self._fn_ = os.path.join(self._tmp_dir.name, FN_JSON)
+        self._cls = JSONDict
+        self._backend_kwargs = {'filename': self._fn_, 'write_concern': self._write_concern}
+        yield self._cls(**self._backend_kwargs)
+        self._tmp_dir.cleanup()
+
+    def store(self, data):
+        with open(self._fn_, 'wb') as file:
+            file.write(json.dumps(data).encode())
+
+    def test_init(self, synced_dict):
+        assert len(synced_dict) == 0
+
+    def test_invalid_kwargs(self):
+        with pytest.raises(ValueError):
+            return self._cls()
+
+    def test_isinstance(self, synced_dict):
+        assert isinstance(synced_dict, SyncedCollection)
+        assert isinstance(synced_dict, MutableMapping)
+        assert isinstance(synced_dict, self._cls)
```

Comment:
As pointed out before, the `self._cls` variable defeats the purpose of using fixtures in the first place.